examples/Potentials and capacities.ipynb

```diff
@@ -0,0 +1,671 @@
dzVE+GGzdu0K1bt3LZ1v79+wkNDdVpnRdffJFJkyaVTUCPISgoiLZt22JtbU2tWrXo1asX0dHRGsucPHmSzZs36ylCIYQQQoiKTyoahBB6M2LECEJCQrh+/XqBecHBwbRp04YWLVroIbLKr3bt2piampbLtmrUqEGNGjXKZVtl7dChQ4wbN45ff/2VkJAQMjMzeemll0hOTlaWsbe3p3r16nqMUgghhBCiYpOKBiGE3rz88svY29uzatUqjelJSUn88MMPjBgxAoDNmzfTrFkzTE1NcXNzY+HChYWWmdec/8yZM8q0hIQEVCoVBw8eBODgwYOoVCr27t1Lq1atMDc3p1OnTsTHx7N7926aNGmCjY0NAwcOJCUlRSknJyeHoKAg3N3dMTc3p2XLlvz444867XNcXBw9e/bEysoKGxsb/Pz8uHXrljI/JiaGnj174uDggJWVFW3btmX//v0aZcTHx9OjRw/Mzc1xd3dn3bp1BbbzcNeJvGOyZcsWOnbsiIWFBS1btuT48eMa66xYsQJnZ2csLCzo3bs3ixYtolq1ajrtH2hvrdCrVy8CAgJ0LitPTk4OH3/8MQ0aNMDU1BQXFxc++OAD4L/927RpE88//zzm5ua0bduWCxcucPLkSdq0aYOVlRXdunXj9u3bRW5nz549BAQE0KxZM1q2bMmqVauIi4sjIiKixLELIYQQQlQ1UtEghNAbIyMjhgwZwqpVq1Cr1cr0H374gezsbAYMGEBERAR+fn689tprnDt3jsDAQGbPnl2gcqIkAgMD+fLLLzl27BjXrl3Dz8+PxYsXs379enbu3Mm+ffv44osvlOWDgoJYs2YNy5Yt4/z580yePJlBgwZx6NChYm0vJyeHnj178s8//3Do0CFCQkK4fPky/fv3V5ZJSkqie/fuhIaGcvr0aXx9fenRowdxcXHKMgEBAVy7do0DBw7w448/8vXXXxMfH//I7b/zzjtMmzaNM2fO4OHhwYABA8jKygIgLCyM0aNH8+abb3LmzBm6du2q3MhXBLNmzWLBggXMnj2byMhI1q9fj4ODg8Yyc+fO5d133+XUqVMYGRkxcOBApk+fzueff86RI0e4dOkSc+bM0Wm79+/fB5AWDEIIIYQQulALIYQeRUVFqQH1gQMHlGnPP/+8etCgQWq1Wq0eOHCgumvXrhrrvPXWW+qmTZsqr11dXdWfffaZWq1Wq2NjY9WA+vTp08r8e/fuaWzjwIEDakC9f/9+ZZmgoCA1oI6JiVGmvfHGG2ofHx+1Wq1Wp6WlqS0sLNTHjh3TiGXEiBHqAQMGFLp/D8e2b98+taGhoTouLk6Zf/78eTWgDg8PL7SMZs2aqb/44gu1Wq1WR0dHF1g+7xjmbUetVqsB9datWzWOyTfffFNgu1FRUWq1Wq3u37+/+n//+5/Gdv39/dW2traFxqXtWKvVarW3t7f6zTff1JjWs2dP9dChQ4tcpjAPHjxQm5qaqlesWFFkHA/v3/fff68G1KGhocq0oKAgdaNGjYq1TbVarc7Ozlb/73//U3fo0KHAvLxz6N69e8UuTwghhBCiqpAWDUIIvWrcuDFeXl6sXLkSgEuXLnHkyBGl20RUVBQdOnTQWKdDhw5cvHiR7Ozsx9r2w/kfHBwcsLCwoF69ehrT8loKXLp0iZSUFLp27YqVlZXys2bNGmJiYoq1vaioKJydnXF2dlamNW3alGrVqhEVFQXktmiYNm0aTZo0oVq1alhZWREVFaW0aIiKisLIyIjWrVsrZTRu3LhYXRwe3t86deoAKPsXHR3NM888o7F8/tf6EhUVRXp6Op07dy5yufzvJ0Dz5s01puXt75EjRzTeR23dT8aNG8cff/zBhg0bSmM3hBBCCCGqDCN9ByCEECNGjGDChAl89dVXBAcHU79+fby9vUtUloFBbv2p+qGuGJmZmVqXNTY2Vv6vUqk0XudNy8nJAXIrAAB27tyJk5OTxnKlmXRx2rRphISE8Omnn9KgQQPMzc159dVXycjIeOyy8+8voOxfaTIwMNA4/lD4e1Ac5ubmxVpO2/7ln5a3v23atNHI45G/G8b48ePZsWMHhw8fpm7duiUNXQghhBCiSpIWDUIIvfPz88PAwID169ezZs0ahg8frtwoNmnShLCwMI3lw8LC8PDwwNDQsEBZ9vb2QO7wjnkevqEsqaZNm2JqakpcXBwNGjTQ+Hm4hUJRmjRpwrVr17h27ZoyLTIykoSEBJo2bQrk7ltAQAC9e/emefPm1K5dmytXrijLN27cmKysLI3khNHR0SQkJDzW/jVq1IiTJ09qTMv/urjs7e01jn92djZ//PFHiWNr2LAh5ubmOg+hWRRzc3ON99Da2hrIraAaP348W7du5ZdffsHd3b3UtimEEEIIUVVIiwYhhN5ZWVnRv39/Zs2axYMHDzRGJ5g6dSpt27Zl/vz59O/fn+PHj/Pll1/y9ddfay3L3NycZ599lgULFuDu7k58fDzvvvvuY8dobW3NtGnTmDx5Mjk5OTz33HPcv3+fsLAwbGxsGDp06CPL6NKlC82bN8ff35/FixeTlZXF2LFj8fb2pk2bNkDuTfWWLVvo0aMHKpWK2bNna7Q6aNSoEb6+vrzxxhssXboUIyMjJk2aVOyn/oWZMGECL7zwAosWLaJHjx788ssv7N69W6nw0UWnTp2YMmUKO3fupH79+ixatEhrRcjt27cLVALVqVOnQOsCMzMzZsyYwfTp0zExMaFDhw7cvn2b8+fPK11sSsu4ceNYv34927dvx9ramps3bwJga2v72MdYCCGEEKKqkBYNQogKYcSIEdy7dw8fHx8cHR2V6Z6enmzatIkNGzbw1FNPMWfOHN57770ih0pcuXIlWVlZtG7dmkmTJvH++++XSozz589n9uzZBAUF0aRJE3x9fdm5c2exn3qrVCq2b9+OnZ0dL7zwAl26dKFevXps3LhRWWbRokXY2dnh5eVFjx498PHxwdPTU6Oc4OBgHB0d8f
```

Comment:
can you explain more cleary in the preceding text the rules for how thickness and N/P ratios are given? my understanding is that currently if you give both thicknesses then the N/P ratio is ignored, otherwise if you give one thickness and an N/P ratio then they are used together to give the other thickness?  

if users provided both thickness and the N/P ratio then we should raise a warning (maybe even an error) saying the N/P ratio will be ignored - does this happen?